modules/framePreProc/framePreProcWorker.py

```diff
@@ -0,0 +1,35 @@
+from framePreProc import FramePreProc
+
+lastTelemetry = None
+
+def worker(telemetryDataObject, pipelineOut):
+
+    """
+
+    Takes Euler data and passes it to the pipeline if the data passes through the framePreProc filter
+
+    Parameters
+    ----------
+
+    telemetryDataObject: MergedData object
+        Includes the telemetry data + euler angles data via .telemetry
+
+    pipelineOut: Queue?
+        Destination of telemetry data successfully passing through the filter
+
+    """
+
+    global lastTelemetry
```

Comment:
Here's how I'd restructure this. I wouldn't maintain lastTelemetry anywhere in the worker function. Instead, first get the eulerAnglesDict from the input data, then pass it into the filter function, then pass it into the update_last_dict function. If it passes the filter, put the input data in the output pipeline. It should look like this.

```
def frame_pre_proc_worker(pipelineIn, pipelineOut):
   frame_proc = FramePreProc()
   while true:
      dataObj = pipelineIn.get()
      eulerAnglesDict = dataObj.telemetry['eulerAnglesOfCamera']
      filter_pass = frame_proc.filter(eulerAnglesDict)
      frame_pre_proc.update_last_dict(eulerAnglesDict)
      
      if filter_pass:
         pipelineOut.put(dataObj)
      else:
         continue
```